rasa/core/domain.py

```diff
@@ -732,10 +753,37 @@ def persist(self, filename: Union[Text, Path]) -> None:
         domain_data = self.as_dict()
         utils.dump_obj_as_yaml_to_file(filename, domain_data)
 
-    def cleaned_domain(self) -> Dict[Text, Any]:
-        """Fetch cleaned domain, replacing redundant keys with default values."""
+    @staticmethod
+    def _transform_intents_for_file(
+        intent_properties: Dict[Text, Dict[Text, Union[bool, List]]],
+        entities: List[Text],
+    ) -> List[Union[Text, Dict[Text, Any]]]:
+        """Replace the internal `used_entities` property by `use_entities` or
+        `ignore_entities`.
+        """
+        intent_properties_for_file = []
+
+        for intent_name, intent_props_internal in intent_properties.items():
+            intent_for_file = {}
+            use_entities = set(intent_props_internal["used_entities"])
+            ignore_entities = set(entities) - use_entities
+            if len(use_entities) == len(entities):
+                intent_for_file["use_entities"] = True
+            elif len(use_entities) <= len(entities) / 2:
+                intent_for_file["use_entities"] = list(use_entities)
+            else:
+                intent_for_file["ignore_entities"] = list(ignore_entities)
+            intent_properties_for_file.append({intent_name: intent_for_file})
+
+        return intent_properties_for_file
 
+    def cleaned_domain(self) -> Dict[Text, Any]:
+        """Fetch cleaned domain, replacing redundant keys with default values and
```

Comment:
optional-ish: I think this is too much `what` and to less `why`. The docstring should describe `why` we are cleaning the domain.